File: core/src/test/scala/com/sksamuel/scrimage/RGBColorTest.scala
package com.sksamuel.scrimage

import com.sksamuel.scrimage.Color._
import org.scalatest.{FlatSpec, OneInstancePerTest}

/** @author Stephen Samuel */
class RGBColorTest extends FlatSpec with OneInstancePerTest {

  "an RGB color" should "convert to hex" in {
    assert("FF00FF" === Color(255, 0, 255).toHex)
    assert("FFFFFF" === Color(255, 255, 255).toHex)
    assert("EFEFEF" === Color(239, 239, 239).toHex)
    assert("00000F" === Color(0, 0, 15).toHex)
    assert("000000" === Color(0, 0, 0).toHex)
  }

  it should "convert to an integer using correct bit shifting" in {
    assert(0xFFEFDFCF === Color(239, 223, 207).argb)
  }

  it should "convert to awt.color" in {
    assert(new java.awt.Color(255, 0, 255) === Color(255, 0, 255).toAWT)
    assert(new java.awt.Color(255, 250, 255) === Color(255, 250, 255).toAWT)
    assert(new java.awt.Color(14, 250, 255) === Color(14, 250, 255).toAWT)
    assert(new java.awt.Color(255, 0, 0) === Color(255, 0, 0).toAWT)
  }

  it should "convert from awt.color" in {
    import Color._
    assert(awt2color(new java.awt.Color(255, 0, 255)) === Color(255, 0, 255))
    assert(awt2color(new java.awt.Color(255, 250, 255)) === Color(255, 250, 255))
    assert(awt2color(new java.awt.Color(14, 250, 255)) === Color(14, 250, 255))
    assert(awt2color(new java.awt.Color(255, 0, 0)) === Color(255, 0, 0))
  }

  it should "convert from int to color using correct bit shifting" in {
    assert(Color(1088511) === Color(16, 155, 255, 0))
  }
}

File: scrimage-core/src/test/kotlin/com/sksamuel/scrimage/core/color/RGBColorTest.kt
package com.sksamuel.scrimage.core.color

import com.sksamuel.scrimage.color.RGBColor
import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe
import java.awt.Color

class RGBColorTest : StringSpec({

   "an RGB color convert to hex" {
      "FF00FF" shouldBe RGBColor(255, 0, 255).toHex()
      "FFFFFF" shouldBe RGBColor(255, 255, 255).toHex()
      "EFEFEF" shouldBe RGBColor(239, 239, 239).toHex()
      RGBColor(0, 0, 15).toHex() shouldBe "00000F"
      RGBColor(0, 0, 0).toHex() shouldBe "000000"
   }

   "convert to an integer using correct bit shifting"  {
      RGBColor(239, 223, 207).toARGBInt() shouldBe -1056817
      RGBColor(239, 223, 207, 0).toARGBInt() shouldBe 0xEFDFCF
   }

   "convert to awt.color"  {
      Color(255, 0, 255) shouldBe RGBColor(255, 0, 255).toAWT()
      Color(255, 250, 255) shouldBe RGBColor(255, 250, 255).toAWT()
      Color(14, 250, 255) shouldBe RGBColor(14, 250, 255).toAWT()
      Color(255, 0, 0) shouldBe RGBColor(255, 0, 0).toAWT()
   }

   "convert from awt.color"  {
      RGBColor.fromAwt(Color(255, 0, 255)) shouldBe RGBColor(255, 0, 255)
      RGBColor.fromAwt(Color(255, 250, 255)) shouldBe RGBColor(255, 250, 255)
      RGBColor.fromAwt(Color(14, 250, 255)) shouldBe RGBColor(14, 250, 255)
      RGBColor.fromAwt(Color(255, 0, 0)) shouldBe RGBColor(255, 0, 0)
   }

   "convert from int to color using correct bit shifting"  {
      RGBColor.fromARGBInt(1088511) shouldBe RGBColor(16, 155, 255, 0)
   }

})